src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IProjectContext.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    internal interface IProjectContext
```

Comment:
But this is only used by an Rolsyn specific adapter in the project system to talk to roslyn host projects, so it still works.

You only really have two choices. Either the interface is a project system interface for pushing information that the project system knows to consumers or it is a Roslyn interface for manipulating a Roslyn type.  All other choices are half measures that cause the meaning of the type to become murky.  

---
In reply to: [71422154](https://github.com/dotnet/roslyn/pull/12440#discussion_r71422154) [](ancestors = 71422154)